File: src/main/java/sql/SqlTest1.java
package sql;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class SqlTest1 {
    public static void main(String[] args) {
        SparkSession ss = SparkSession.builder().master("local").appName("sql").config("spark.some.config.option", "some-value").getOrCreate();
        Dataset<Row> ds = ss.read().format("csv")
                .option("sep", ",")
                .option("inferSchema", "true")
                .option("header", "true")
                .load("data.csv");
        ds.select("name").show();
    }
}

File: src/main/java/sql/SqlTest1.scala
package sql

import org.apache.spark.sql.{DataFrame, SparkSession}


/**
 * 由文件数据创建df
 */
object SqlTest1 {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder().master("local[*]").appName("sql-test").getOrCreate()
    spark.sparkContext.setLogLevel("WARN")
    val data1: DataFrame = spark.read.csv("./data.csv")
    data1.show()
    data1.select("_c0").show()
    spark.stop()

  }
}
